lib/cdo/rack/whitelist.rb

```diff
@@ -0,0 +1,107 @@
+# Rack middleware that whitelists cookies and headers based on path-based cache behaviors.
+# Behaviors are defined in http cache config.
```

Comment:
Could you add a design comment or a link to a design doc about how the X-Cookie headers are used?